crates/sargon/src/system/sargon_os/sargon_os.rs

```diff
@@ -151,6 +144,51 @@ impl SargonOS {
         Ok(())
     }
 
+    pub async fn new_wallet_with_derived_bdfs(
+        &self,
+        hd_factor_source: PrivateHierarchicalDeterministicFactorSource,
+        accounts: Accounts,
+    ) -> Result<()> {
+        debug!("Deriving Profile from BDFS");
+
+        let hd_keys: Vec<HierarchicalDeterministicPublicKey> = accounts
+            .iter()
+            .map(|account| {
+                account
+                    .security_state
+                    .into_unsecured()
+                    .map(|c| c.transaction_signing.public_key)
+                    .map_err(|_| CommonError::EntitiesNotDerivedByFactorSource)
+            })
+            .try_collect()?;
+
+        if !hd_factor_source
+            .mnemonic_with_passphrase
+            .validate_public_keys(hd_keys)
```

Comment:
nice! so now we assert that all factor instances are indeed controlled by the specified mnemonic